### ISSUE TITLE
Is the code complete?

### ISSUE THREAD

**Issue by Moshu413** [ziqin-h/GIVEPose#6]
Thanks for your work! Is the code you provided complete? I found that the performance metrics were quite lacking after trying to replicate it. Looking forward to your reply!

**Comment by ziqin-h** [comment]
Could you specify what exactly you mean by the metrics being "lacking"? The code we provided should be complete. Here are some suggestions: You can first try testing with the provided checkpoint to ensure you have a correct understanding of the test script. If you have conducted training, you can check if the training loss is normal first.

**Comment by Moshu413** [comment]
Thank you for your response. I downloaded the checkpoint you provided and ran a test using the following command:

python evaluation/evaluate.py --resume_model="./output/model_givepose.pth" --dataset=Real --use_scale_net --sn_path='./output/scale_net/model_scale_net_Real.pth'

However, the final output results were as follows:2025-08-11 00:05:05,266 : average mAP:
2025-08-11 00:05:05,266 : 3D IoU at 25: 41.8
2025-08-11 00:05:05,267 : 3D IoU at 50: 18.9
2025-08-11 00:05:05,267 : 3D IoU at 75: 3.2
2025-08-11 00:05:05,267 : 5 degree, 5cm: 8.2
2025-08-11 00:05:05,267 : 10 degree, 5cm: 15.1
2025-08-11 00:05:05,267 : 10 degree, 10cm: 34.4
2025-08-11 00:05:05,267 : 10 degree: 52.4
2025-08-11 00:05:05,267 : 10cm: 46.1
2025-08-11 00:05:05,267 : category bottle
2025-08-11 00:05:05,267 : mAP:
2025-08-11 00:05:05,268 : 3D IoU at 25: 0.0
2025-08-11 00:05:05,268 : 3D IoU at 50: 0.0
2025-08-11 00:05:05,268 : 3D IoU at 75: 0.0
2025-08-11 00:05:05,268 : 5 degree, 5cm: 0.0
2025-08-11 00:05:05,268 : 10 degree, 5cm: 0.0
2025-08-11 00:05:05,268 : 10 degree, 10cm: 0.1
2025-08-11 00:05:05,268 : category bowl
2025-08-11 00:05:05,268 : mAP:
2025-08-11 00:05:05,269 : 3D IoU at 25: 53.2
2025-08-11 00:05:05,269 : 3D IoU at 50: 23.4
2025-08-11 00:05:05,269 : 3D IoU at 75: 8.0
2025-08-11 00:05:05,269 : 5 degree, 5cm: 23.5
2025-08-11 00:05:05,269 : 10 degree, 5cm: 23.7
2025-08-11 00:05:05,269 : 10 degree, 10cm: 55.1
2025-08-11 00:05:05,269 : category camera
2025-08-11 00:05:05,269 : mAP:
2025-08-11 00:05:05,270 : 3D IoU at 25: 44.2
2025-08-11 00:05:05,270 : 3D IoU at 50: 12.3
2025-08-11 00:05:05,270 : 3D IoU at 75: 0.6
2025-08-11 00:05:05,270 : 5 degree, 5cm: 0.0
2025-08-11 00:05:05,270 : 10 degree, 5cm: 4.6
2025-08-11 00:05:05,270 : 10 degree, 10cm: 17.3
2025-08-11 00:05:05,270 : category can
2025-08-11 00:05:05,271 : mAP:
2025-08-11 00:05:05,271 : 3D IoU at 25: 31.6
2025-08-11 00:05:05,271 : 3D IoU at 50: 16.9
2025-08-11 00:05:05,271 : 3D IoU at 75: 3.9
2025-08-11 00:05:05,271 : 5 degree, 5cm: 13.1
2025-08-11 00:05:05,271 : 10 degree, 5cm: 32.5
2025-08-11 00:05:05,271 : 10 degree, 10cm: 61.4
2025-08-11 00:05:05,271 : category laptop
2025-08-11 00:05:05,272 : mAP:
2025-08-11 00:05:05,272 : 3D IoU at 25: 83.7
2025-08-11 00:05:05,272 : 3D IoU at 50: 47.0
2025-08-11 00:05:05,272 : 3D IoU at 75: 6.8
2025-08-11 00:05:05,272 : 5 degree, 5cm: 5.9
2025-08-11 00:05:05,272 : 10 degree, 5cm: 8.9
2025-08-11 00:05:05,272 : 10 degree, 10cm: 33.3
2025-08-11 00:05:05,272 : category mug
2025-08-11 00:05:05,272 : mAP:
2025-08-11 00:05:05,273 : 3D IoU at 25: 38.1
2025-08-11 00:05:05,273 : 3D IoU at 50: 14.1
2025-08-11 00:05:05,273 : 3D IoU at 75: 0.0
2025-08-11 00:05:05,273 : 5 degree, 5cm: 6.5
2025-08-11 00:05:05,273 : 10 degree, 5cm: 20.9
2025-08-11 00:05:05,273 : 10 degree, 10cm: 39.1
2025-08-11 00:05:57,505 : average mAP:
2025-08-11 00:05:57,505 : 3D IoU at 25: 55.8
2025-08-11 00:05:57,505 : 3D IoU at 50: 38.8
2025-08-11 00:05:57,505 : 3D IoU at 75: 19.1
2025-08-11 00:05:57,505 : 5 degree, 20%: 22.2
2025-08-11 00:05:57,505 : 10 degree, 20%: 35.2
2025-08-11 00:05:57,506 : 10 degree, 50%: 53.8
2025-08-11 00:05:57,506 : 10 degree: 61.9
2025-08-11 00:05:57,506 : 20%: 37.9
2025-08-11 00:05:57,506 : 50%: 65.0
2025-08-11 00:05:57,506 : category bottle
2025-08-11 00:05:57,506 : mAP:
2025-08-11 00:05:57,506 : 3D IoU at 25: 26.6
2025-08-11 00:05:57,506 : 3D IoU at 50: 4.6
2025-08-11 00:05:57,507 : 3D IoU at 75: 0.5
2025-08-11 00:05:57,507 : 5 degree, 20%: 2.9
2025-08-11 00:05:57,507 : 10 degree, 20%: 4.6
2025-08-11 00:05:57,507 : 10 degree, 50%: 39.6
2025-08-11 00:05:57,507 : 10 degree: 49.7
2025-08-11 00:05:57,507 : 20%: 4.8
2025-08-11 00:05:57,507 : 50%: 43.5
2025-08-11 00:05:57,507 : category bowl
2025-08-11 00:05:57,508 : mAP:
2025-08-11 00:05:57,508 : 3D IoU at 25: 100.0
2025-08-11 00:05:57,508 : 3D IoU at 50: 98.9
2025-08-11 00:05:57,508 : 3D IoU at 75: 73.5
2025-08-11 00:05:57,508 : 5 degree, 20%: 77.6
2025-08-11 00:05:57,508 : 10 degree, 20%: 97.9
2025-08-11 00:05:57,508 : 10 degree, 50%: 99.9
2025-08-11 00:05:57,508 : 10 degree: 99.9
2025-08-11 00:05:57,508 : 20%: 98.1
2025-08-11 00:05:57,509 : 50%: 100.0
2025-08-11 00:05:57,509 : category camera
2025-08-11 00:05:57,509 : mAP:
2025-08-11 00:05:57,509 : 3D IoU at 25: 60.2
2025-08-11 00:05:57,509 : 3D IoU at 50: 26.3
2025-08-11 00:05:57,509 : 3D IoU at 75: 1.3
2025-08-11 00:05:57,509 : 5 degree, 20%: 0.8
2025-08-11 00:05:57,509 : 10 degree, 20%: 10.2
2025-08-11 00:05:57,510 : 10 degree, 50%: 27.8
2025-08-11 00:05:57,510 : 10 degree: 30.8
2025-08-11 00:05:57,510 : 20%: 21.1
2025-08-11 00:05:57,510 : 50%: 69.8
2025-08-11 00:05:57,510 : category can
2025-08-11 00:05:57,510 : mAP:
2025-08-11 00:05:57,510 : 3D IoU at 25: 61.3
2025-08-11 00:05:57,510 : 3D IoU at 50: 32.8
2025-08-11 00:05:57,510 : 3D IoU at 75: 6.1
2025-08-11 00:05:57,511 : 5 degree, 20%: 21.8
2025-08-11 00:05:57,511 : 10 degree, 20%: 41.7
2025-08-11 00:05:57,511 : 10 degree, 50%: 86.7
2025-08-11 00:05:57,511 : 10 degree: 90.4
2025-08-11 00:05:57,511 : 20%: 41.9
2025-08-11 00:05:57,511 : 50%: 88.7
2025-08-11 00:05:57,511 : category laptop
2025-08-11 00:05:57,512 : mAP:
2025-08-11 00:05:57,512 : 3D IoU at 25: 85.4
2025-08-11 00:05:57,512 : 3D IoU at 50: 70.2
2025-08-11 00:05:57,512 : 3D IoU at 75: 33.2
2025-08-11 00:05:57,512 : 5 degree, 20%: 30.2
2025-08-11 00:05:57,512 : 10 degree, 20%: 56.6
2025-08-11 00:05:57,512 : 10 degree, 50%: 66.6
2025-08-11 00:05:57,512 : 10 degree: 67.2
2025-08-11 00:05:57,513 : 20%: 61.8
2025-08-11 00:05:57,513 : 50%: 85.4
2025-08-11 00:05:57,513 : category mug
2025-08-11 00:05:57,513 : mAP:
2025-08-11 00:05:57,513 : 3D IoU at 25: 1.0
2025-08-11 00:05:57,513 : 3D IoU at 50: 0.0
2025-08-11 00:05:57,513 : 3D IoU at 75: 0.0
2025-08-11 00:05:57,513 : 5 degree, 20%: 0.0
2025-08-11 00:05:57,514 : 10 degree, 20%: 0.0
2025-08-11 00:05:57,514 : 10 degree, 50%: 2.5
2025-08-11 00:05:57,514 : 10 degree: 33.2
2025-08-11 00:05:57,514 : 20%: 0.0
2025-08-11 00:05:57,514 : 50%: 2.6
I’m not sure where I went wrong—the output results don’t match those in the paper. I’m still learning in this field, so I truly appreciate your help and response.

**Comment by ziqin-h** [comment]
I compared the test results you provided with my local test results, and it seems that all indicators of other categories are consistent except for the "mug" category. Therefore, you may need to check where the problem lies with the mug category. For example, we used the refine_for_mug segmentation, etc.
[https://github.com/ziqin-h/GIVEPose/blob/653fe1a2d3316e6748ca377d475731c97c1bf711/evaluation/load_data_eval.py#L39](url)

**Comment by Moshu413** [comment]
Thanks for the reply, this should be where the problem occurred, I downloaded 'segmentation_results' from 'DualPose-Net' and put the directory as follows. But I can't find where the 'segmentation_results_refine_for_mug' file is, where can I get it?

<img width="411" height="161" alt="Image" src="https://github.com/user-attachments/assets/99058f57-9213-40f9-9036-bc85456cf914" />

**Comment by ziqin-h** [comment]
You can get this file from https://drive.google.com/drive/folders/1q8pjmHDfSUTna13F2R_gU3P-FYCjEP7A

**Comment by Moshu413** [comment]
> You can get this file from https://drive.google.com/drive/folders/1q8pjmHDfSUTna13F2R_gU3P-FYCjEP7A

Thanks for your reply!
